src/GraphQL.Tests/Validation/ArgumentsOfCorrectTypeTests.cs

```diff
@@ -151,7 +151,7 @@ public void float_into_string()
             ShouldFailRule(_ =>
             {
                 _.Query = query;
-                Rule.badValue(_, "stringArg", "String", "1.0", 3, 32);
+                Rule.badValue(_, "stringArg", "String", "1", 3, 32);
```

Comment:
Original query contains `1.0`. It will be incorrectly to show another value in the error message. The question is completely different. Why does `IsValidLiteralValue` call a `AstPrinter.Print` method? `valueAst` is `INode` and contains `SourceLocation` (start, end) so to get the required text we just need to call `originalquery.Subscting(valueAst.SourceLocation.Start, valueAst.SourceLocation.End - valueAst.SourceLocation.Start)`. Nothing to "print" here. But where to take `originalquery` 😄 ? Look at `ISchema schema`. It is not used 😮 ! Just change it to `ValidationContext`. There are only two usages from validation rules.

Reply:
Is the originalQuery always provided? Or is it optional if document is provided? I think this demonstrates a flaw in the design of the AST nodes - they should not be deserialized and in fact should not be different than the ones provided by parser. Separate discussion of course.

Reply:
Anyway it would seem you’ve found a separate bug, which could be reproduced without these changes. For example if the number was 2E+0 then it would have printed 2.0 before which would also be incorrect.

Reply:
> Is the originalQuery always provided? Or is it optional if document is provided?

Yes - `GraphQLDocumentBuilder.Build`. There is no more separate `OriginalQuery` property since #2276.

Reply:
The issue you noted with the error messages is fixed in #2397

Reply:
ok